autoload/lsp.vim

```diff
@@ -214,7 +214,11 @@ function! s:call_did_save(buf, server_name, result, cb) abort
 endfunction
 
 function! s:on_text_document_did_close() abort
-    call lsp#log('s:on_text_document_did_close()', bufnr('%'))
+    let l:buf = bufnr('%')
+    call lsp#log('s:on_text_document_did_close()', l:buf)
+    if has_key(s:file_content, l:buf)
```

Comment:
`vim-lsp` also support multiple language servers for the same filetype. For example you could have two different language server - `typescript-langauge-server` and `tslint language server` running for typescript files. One may support incremental changes one may not. Or one could be started earlier, one could be started late or stopped manually by the user so diff's are not guaranteed to be the same.

I think the structure needs to look something like this.

```vim
let s:file_content = {
    '1': { " bufnr
        'typescript-language-server': [], " diff
        'tslint-language-server': []      " diff
    }
}
```

While not required it would also be good to cleanup if server exists instead of just buffer getting close.